File: src/main/scala/io/sdkman/DbMigration.scala
package io.sdkman

import com.github.mongobee.Mongobee

object DbMigration extends App with Configuration {

  mongoUsernameO
    .fold(localMongoConn)(mongoUsername => remoteMongoConn(mongoUsername))
    .setDbName(mongoDatabase)
    .setChangeLogsScanPackage("io.sdkman.changelogs")
    .execute()

  lazy val localMongoConn = mongobee(s"mongodb://$mongoHost:$mongoPort")

  def remoteMongoConn(username: String) =
    mongobee(
      s"mongodb://$username:$mongoPassword@$mongoHost:$mongoPort/$mongoDatabase?authMechanism=SCRAM-SHA-1"
    )

  private def mongobee(url: String) = new Mongobee(url)
}

File: src/main/java/io/sdkman/DbMigration.java
package io.sdkman;

import com.github.mongobee.Mongobee;
import com.github.mongobee.exception.MongobeeException;

public class DbMigration {
    public static void main(String[] args) throws MongobeeException {
        Mongobee runner = new Mongobee("mongodb://scopuli:27017/sdkman");
        runner.setDbName("sdkman");
        runner.setChangeLogsScanPackage("io.sdkman.changelogs");
        runner.execute();
        System.out.println("Run db migration.");
    }
}
